onnxruntime/test/shared_lib/test_allocator.cc

```diff
@@ -31,4 +31,8 @@ TEST(CApiTest, DefaultAllocator) {
   ASSERT_EQ(allocation.size(), 100U);
   ASSERT_NE(allocation.get(), nullptr);
   memset(allocation.get(), 0, 100U);
+
+  // Default Allocator has no stats to report
+  auto stats = default_allocator.GetStats();
+  ASSERT_EQ(stats.size(), 0U);
 }
```

Comment:
Please, provide test coverage for both test cases, when the allocator has stats and when it does not.

Check for Ort::Status returned.

For the allocator with stats you can query either Arena Allocator or create your own artificial allocator that reports stats.

Reply:
Sure. I used to add the GetStats check in `CApiTest::get_allocator_cpu`. Strangely, it failed in some CI tests and seemed that the queried allocator does not have a GetStats implementation. But the tests works ok in my local machine. I'll add the type info in the error message and troubleshoot the issue.

Reply:
I added tests for GetStats in the following test cases:
- CApiTest::DefaultAllocator: Return ORT_NOT_IMPLEMENTED
- CApiTest::get_allocator_cpu:
  - Check both keys and values if it supports arena usage
  - Return ORT_NOT_IMPLEMENTED otherwise.
- CApiTest::get_allocator_cuda: Check both keys and values.

Pls check if this is sufficient.